### PR TITLE
[#84] Activate tests in `Test.Cardano.Chain.Genesis.Json` & `Test.Cardano.Chain.Txp.Json`

### LINKED ISSUE #84: Activate currently inactive tests
Currently there are tests in `Test.Cardano.Chain.Genesis.Json` & `Test.Cardano.Chain.Txp.Json` that are not running. This issue brings the necessary types from the Byron release to make these tests run.

### PR DESCRIPTION
**Pull request by Jimbo4350**
Currently the tests in `Test.Cardano.Chain.Genesis.Json` & `Test.Cardano.Chain.Txp.Json` are not running. This PR brings the necessary types from the Byron release for the tests to run.

Closes #84

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ packages:
 
 extra-deps:
   - git: https://github.com/input-output-hk/cardano-prelude
-    commit: 4365aeb5d5bec9c47083639cc06784e654fefe6d
+    commit: c219477a4313767d9786147008026d84afce2f12
```
Comment (Jimbo4350) [inline stack.yaml:13]: Updated commit to import `goldenTestJSONPretty`

File: cardano-chain.cabal [hunk 2]
```diff
@@ -111,7 +119,6 @@ library
                      , cryptonite
                      , Cabal
                      , formatting
-                     , hashable
```
Comment (ruhatch) [inline cardano-chain.cabal:114]: :+1:

File: src/Cardano/Chain/Genesis/AvvmBalances.hs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import           Data.Aeson (FromJSON (..), ToJSON (..))
+import qualified Data.Map.Strict as M
+
+import           Cardano.Chain.Common (Coin)
+import           Cardano.Crypto.Signing.Redeem (RedeemPublicKey)
+
+-- | Predefined balances of avvm entries.
+newtype GenesisAvvmBalances = GenesisAvvmBalances
+    { getGenesisAvvmBalances :: M.Map RedeemPublicKey Coin
```
Comment (ruhatch) [inline src/Cardano/Chain/Genesis/AvvmBalances.hs]: `Map` is exposed by `Cardano.Prelude`, so you don't need this `M` here. If you're not using the operations from `Data.Map.Strict` you don't need the import at all.

File: src/Cardano/Chain/Genesis/Delegation.hs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+mkGenesisDelegation psks = do
+    unless ((length . nub $ pskIssuerPk <$> psks) == length psks) $
+        throwError "all issuers must be distinct"
+    let res = M.fromList [(StakeholderId $ addressHash (pskIssuerPk psk), psk) | psk <- psks]
+    recreateGenesisDelegation res
+
+-- | Safe constructor of 'GenesisDelegation' from existing map.
+recreateGenesisDelegation
+    :: MonadError Text m
+    => M.Map StakeholderId ProxySKHeavy
```
Comment (ruhatch) [inline src/Cardano/Chain/Genesis/Delegation.hs]: Same thing here!

File: test/Test/Cardano/Chain/Common/Example.hs [hunk 5]
```diff
@@ -1,21 +1,100 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -fno-warn-incomplete-uni-patterns #-}
```
Comment (ruhatch) [inline test/Test/Cardano/Chain/Common/Example.hs:3]: I like to leave an empty line between the `LANGUAGE` and `OPTIONS_GHC` sections

File: test/Test/Cardano/Chain/Genesis/Example.hs [hunk 6]
```diff
@@ -50,12 +60,12 @@ exampleStaticConfig_GCSpec =
 exampleGenesisAvvmBalances :: GenesisAvvmBalances
 exampleGenesisAvvmBalances =
     GenesisAvvmBalances
-        { getGenesisAvvmBalances = HM.fromList
+        { getGenesisAvvmBalances = M.fromList
             [ ( exampleRedeemPublicKey' (0, 32)
-              , Coin {getCoin = 36524597913081152}
+              , mkKnownCoin @36524597913081152
```
Comment (ruhatch) [inline test/Test/Cardano/Chain/Genesis/Example.hs]: Cool, huh!

File: test/Test/Cardano/Chain/Genesis/Gen.hs [hunk 7]
```diff
@@ -6,19 +6,22 @@ module Test.Cardano.Chain.Genesis.Gen
...
        ) where
 
 import           Cardano.Prelude
+import           Test.Cardano.Prelude
 
 import           Data.Coerce (coerce)
-import qualified Data.HashMap.Strict as HM
-
+import qualified Data.Map.Strict as M
+import           Data.Text (pack)
```
Comment (ruhatch) [inline test/Test/Cardano/Chain/Genesis/Gen.hs]: You can use `toS` for this and avoid the extra `text` dep

File: test/cardano-chain-test.cabal [hunk 8]
```diff
@@ -24,17 +24,23 @@ library
...
                        Test.Cardano.Chain.Txp.Example
                        Test.Cardano.Chain.Txp.Gen
+                       Test.Cardano.Chain.Txp.Json
                        Test.Cardano.Chain.Update.Example
                        Test.Cardano.Chain.Update.Gen
 
   build-depends:       base
                      , base16-bytestring
                      , binary
+                     , bytestring
```
Comment (ruhatch) [inline test/cardano-chain-test.cabal]: Do we need this?

**Review by ruhatch — APPROVED** [review]
Looks good. Just a few little things to consider.

Also we're using 2 spaces instead of 4 for indenting, but if that's a massive hassle I can do another PR for it.

File: test/cardano-chain-test.cabal (hunk 8 above)
Comment (Jimbo4350) [reply test/cardano-chain-test.cabal]: Nope!

**Pull request merged**